src/reconstruct/helper.ts

```diff
@@ -1,20 +1,69 @@
-import { NodeId, ParentInformation, RNode, RType } from '../r-bridge';
-import { ReconstructionConfiguration } from './reconstruct';
+import { NodeId, ParentInformation, RNode, RType } from '../r-bridge'
+import { ReconstructionConfiguration } from './reconstruct'
 
 
 /*
 --helper function--
 */
 export type Selection = Set<NodeId>
+interface PrettyPrintLinePart {
+	part:   string
+	spaces: number
+}
 export interface PrettyPrintLine {
-	line: string
+	line:   string
 	indent: number
 }
-export function plain(text: string): PrettyPrintLine[] {
-	return [{ line: text, indent: 0 }]
+export interface WrappedPrettyPrintLine { //change name to be more fitting
+	linePart: PrettyPrintLinePart[]
+	indent:   number
+}
+export function plain(text: string,space: number): Code {
+	return [unwrapPrettyPrintLine({linePart: [{part: text,spaces: space}],indent: 0})]
 }
 export type Code = PrettyPrintLine[]
 
+export function unwrapPrettyPrintLine(wrappedLine: WrappedPrettyPrintLine): PrettyPrintLine {
+	const text = ''
+	for(let index = 0; index < wrappedLine.linePart.length; index++) {
+		const element = wrappedLine.linePart[index]
+		for(let indent = 0; indent < wrappedLine.linePart[index].spaces; indent++) {
+			text === text + ' '
+		}
```

Comment:
Two things. 
1. First, you should stay consistent and use `element` instead of accessing `linePart` by index again :D

2. Second: With the three equal signs you are no longer assigning, you are actually comparing if `text` is [strict equal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) to `text + ' '` (something which shouldn't be satisfiable :D).

    If you just want to generate a string of `element.spaces` space elements, have a look at `getIndentString` which should do the job with the help of repeat:
    
    ```ts
    function getIndentString(indent: number): string {
	    return ' '.repeat(indent * 4)
    }
    ```
    
    You probably do not want the `* 4` (and/or make this configurable), but to my knowledge this is the fastest way :)